src/Frontend/src/stores/ServiceControlStore.ts

```diff
@@ -0,0 +1,191 @@
+  const isMonitoringEnabled = computed(() => !isMonitoringDisabled.value);
+
+  function getServiceControlUrl() {
+    if (!serviceControlUrl.value) refresh();
+    return serviceControlUrl.value;
+  }
+
+  function getMonitoringUrl() {
+    if (!monitoringUrl.value) refresh();
+    return monitoringUrl.value;
+  }
+
+  function refresh() {
+    const params = getParams();
+    const scu = getParameter(params, "scu");
+    const mu = getParameter(params, "mu");
+
+    if (scu) {
+      serviceControlUrl.value = scu.value;
+      window.localStorage.setItem("scu", serviceControlUrl.value);
+      console.debug(`ServiceControl Url found in QS and stored in local storage: ${serviceControlUrl.value}`);
+    } else if (window.localStorage.getItem("scu")) {
+      serviceControlUrl.value = window.localStorage.getItem("scu");
+      console.debug(`ServiceControl Url, not in QS, found in local storage: ${serviceControlUrl.value}`);
+    } else if (window.defaultConfig && window.defaultConfig.service_control_url) {
+      serviceControlUrl.value = window.defaultConfig.service_control_url;
+      console.debug(`setting ServiceControl Url to its default value: ${window.defaultConfig.service_control_url}`);
+    } else {
+      console.warn("ServiceControl Url is not defined.");
+    }
+
+    if (mu) {
+      monitoringUrl.value = mu.value;
+      window.localStorage.setItem("mu", monitoringUrl.value);
+      console.debug(`Monitoring Url found in QS and stored in local storage: ${monitoringUrl.value}`);
+    } else if (window.localStorage.getItem("mu")) {
+      monitoringUrl.value = window.localStorage.getItem("mu");
+      console.debug(`Monitoring Url, not in QS, found in local storage: ${monitoringUrl.value}`);
+    } else if (window.defaultConfig && window.defaultConfig.monitoring_urls && window.defaultConfig.monitoring_urls.length) {
+      monitoringUrl.value = window.defaultConfig.monitoring_urls[0];
+      console.debug(`setting Monitoring Url to its default value: ${window.defaultConfig.monitoring_urls[0]}`);
+    } else {
+      console.warn("Monitoring Url is not defined.");
+    }
+  }
+
+  async function fetchFromServiceControl(suffix: string, options?: { cache?: RequestCache }) {
+    const requestOptions: RequestInit = {
+      method: "GET",
+      cache: options?.cache ?? "default", // Default  if not specified
+      headers: {
+        Accept: "application/json",
+      },
+    };
+    return await fetch(`${getServiceControlUrl()}${suffix}`, requestOptions);
+  }
+
+  async function fetchTypedFromServiceControl<T>(suffix: string): Promise<[Response, T]> {
+    const response = await fetch(`${getServiceControlUrl()}${suffix}`);
+    if (!response.ok) throw new Error(response.statusText ?? "No response");
+    const data = await response.json();
+
+    return [response, data];
+  }
+
+  async function fetchTypedFromMonitoring<T>(suffix: string): Promise<[Response?, T?]> {
+    if (isMonitoringDisabled.value) {
+      return [];
+    }
```

Comment:
In other parts of the code the caller is enforcing whether it should call the function or not, I think we should do the same here.

```suggestion
```